### PR TITLE
CSVs need to be zipped when emailed

### PR DESCRIPTION
**Pull request by smn**
For some reason CSV attachments can get folded at 1000 chars, making them hard to parse. The hope is that zipping it will prevent that from happening.

### CODE REVIEW

File: go/contacts/tasks.py [hunk 1]
```diff
@@ -83,8 +84,8 @@ def export_group_contacts(account_key, group_key, include_extra):
 
     # collect all names that we can export
     fields = ['name', 'surname', 'email_address', 'msisdn', 'dob',
-                    'twitter_handle', 'facebook_id', 'bbm_pin', 'gtalk_id',
-                    'created_at']
+              'twitter_handle', 'facebook_id', 'bbm_pin', 'gtalk_id',
+              'created_at']
```
Comment (smn) [inline go/contacts/tasks.py:88]: pep8 fix

File: go/contacts/tasks.py [hunk 2]
```diff
@@ -95,14 +96,13 @@ def export_group_contacts(account_key, group_key, include_extra):
 
     # write the CSV header
     extra_fields = sorted(extra_fields)
-    writer.writerow(fields + ['extras-%s' % (key,)
-                                    for key in extra_fields])
+    writer.writerow(fields + ['extras-%s' % (key,) for key in extra_fields])
```
Comment (smn) [inline go/contacts/tasks.py:99]: pep8 fix

File: go/contacts/tasks.py [hunk 3]
```diff
@@ -95,14 +96,13 @@ def export_group_contacts(account_key, group_key, include_extra):
...
 
     # loop over the contacts and create the row populated with
     # the values of the selected fields.
     for contact in contacts:
         row = [unicode(getattr(contact, field, None) or '')
-                for field in fields]
+               for field in fields]
```
Comment (smn) [inline go/contacts/tasks.py:105]: pep8 fix

File: go/contacts/tasks.py [hunk 4]
```diff
@@ -110,12 +110,18 @@ def export_group_contacts(account_key, group_key, include_extra):
...
+
+    email = EmailMessage(
+        '%s contacts export' % (group.name,),
+        'Please find the CSV data for %s contact(s) from '
+        'group "%s" attached.\n\n' % (len(contact_keys), group.name),
         settings.DEFAULT_FROM_EMAIL, [user_profile.user.email])
-    email.attach('contacts-export.csv', io.getvalue(), 'text/csv')
+
+    zipio.seek(0)
+    email.attach('contacts-export.zip', zipio.read(), 'application/zip')
```
Comment (hodgestar) [inline go/contacts/tasks.py]: Is there a reason we do a `.seek(0)` and `.read()` here rather than just doing `.getvalue()`?

**Comment by hodgestar** [comment]
:+1: other than the minor stylistic query to show I read it. :)

File: go/contacts/tasks.py (hunk 4 above)
Comment (smn) [inline go/contacts/tasks.py]: `getvalue()` can only be used before `close()` is called: http://docs.python.org/2/library/stringio.html#StringIO.StringIO.getvalue

Comment (hodgestar) [inline go/contacts/tasks.py]: Hmm. `close()` doc string says _any operations_ after calling `close()` are bad. Maybe we shouldn't use `with` then (since that closes the file)?

**Comment by hodgestar** [comment]
:+1:
